packages/http/src/validator/__tests__/functional.spec.ts

```diff
@@ -31,6 +31,70 @@ describe('HttpValidator', () => {
       });
     });
 
+    describe('deprecated keyword validation', () => {
+      const resource: IHttpOperation = {
+        id: "abc",
+        method: "get",
+        path: "/test",
+        responses: [
+          {
+            code: "200"
+          }
+        ],
+        request: {
+          query: [
+            {
+              style: HttpParamStyles.Form,
+              deprecated: true,
+              name: "productId"
+            }
+          ],
+        },
+      };
+
+      it('returns warnings', () => {
+        expect(
+          validateInput({
+            resource,
+            element: {
+              method: "get",
+              url: {
+                path: "/test",
+                query: {
+                  productId: "abc"
+                }
+              },
+            },
+          }),
+        ).toEqual([
+          {
+            code: "deprecated",
+            message: "Query param productId is deprecated",
+            path: [
+              "query",
+              "productId"
+            ],
+            severity: DiagnosticSeverity.Warning
+          }
+        ]);
+      });
+
+      it('does not return warnings', () => {
```

Comment:
```suggestion
      it('does not return warnings if the property is not supplied', () => {
```

Reply:
@XVincentX, ^ this is explained in the `do` portion of the test - https://github.com/stoplightio/prism/pull/721/files/141d4efe845c8ed9f0a888f74f020103c7f806cf#diff-d79142abfe6fcda19641e6dc5c0aebe4R90. `it` only describes what happens